zipkin-collector/kafka10/src/main/java/zipkin/collector/kafka10/KafkaCollectorWorker.java

```diff
@@ -0,0 +1,114 @@
+/**
+ * Copyright 2015-2017 The OpenZipkin Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package zipkin.collector.kafka10;
+
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.List;
+import java.util.Properties;
+import java.util.concurrent.atomic.AtomicReference;
+import org.apache.kafka.clients.consumer.Consumer;
+import org.apache.kafka.clients.consumer.ConsumerRebalanceListener;
+import org.apache.kafka.clients.consumer.ConsumerRecords;
+import org.apache.kafka.clients.consumer.KafkaConsumer;
+import org.apache.kafka.common.TopicPartition;
+import org.apache.kafka.common.errors.InterruptException;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import zipkin.Codec;
+import zipkin.collector.Collector;
+import zipkin.collector.CollectorMetrics;
+
+import static zipkin.storage.Callback.NOOP;
+
+/** Consumes spans from Kafka messages, ignoring malformed input */
+final class KafkaCollectorWorker implements Runnable {
+  private static final Logger LOG = LoggerFactory.getLogger(KafkaCollectorWorker.class);
+
+  final Consumer<byte[], byte[]> kafkaConsumer;
+  final Collector collector;
+  final CollectorMetrics metrics;
+  final AtomicReference<List<TopicPartition>> assignedPartitions =
+      new AtomicReference<>(Collections.emptyList());
+
+  KafkaCollectorWorker(Properties kafkaConsumerConfig, String topic, Collector collector,
+      CollectorMetrics metrics) {
+    kafkaConsumer = new KafkaConsumer<>(kafkaConsumerConfig);
+    kafkaConsumer.subscribe(Collections.singleton(topic), new ConsumerRebalanceListener() {
+      @Override public void onPartitionsRevoked(Collection<TopicPartition> partitions) {
+        assignedPartitions.set(Collections.emptyList());
+      }
+
+      @Override public void onPartitionsAssigned(Collection<TopicPartition> partitions) {
+        assignedPartitions.set(Collections.unmodifiableList(new ArrayList<>(partitions)));
+      }
+    });
+    this.collector = collector;
+    this.metrics = metrics;
+  }
+
+  /**
+   * @return Kafka topic partitions currently assigned to this worker for processing. Returned
+   * list is not modifiable.
+   */
+  public List<TopicPartition> assignedPartitions() {
```

Comment:
doesn't need to be public. you can use the package protected field for testing. that lets you move this javadoc to the field itself.